mongomock/aggregate.py

```diff
@@ -256,6 +259,38 @@ def _handle_comparison_operator(self, operator, values):
             'aggregation pipeline, it is currently not implemented '
             ' in Mongomock.' % operator)
 
+    def _handle_string_operator(self, operator, values):
+        if operator == '$toLower':
+            parsed = self.parse(values)
+            return str(parsed).lower() if parsed is not None else ''
+        if operator == '$toUpper':
+            parsed = self.parse(values)
+            return str(parsed).upper() if parsed is not None else ''
+        if operator == '$concat':
+            return ''.join([str(self.parse(value)) for value in values])
```

Comment:
What if one of those subvalue is None? would it crash? Please add a test for this.

Reply:
You're right - that's another case. According to docs this will return a concat result of None. Fixed and added test.